sendgrid/sendgrid.py

```diff
@@ -73,6 +78,17 @@ def _get_default_headers(self):
     def reset_request_headers(self):
         self.client.request_headers = self._get_default_headers()
 
+    def attempt(self, request_obj):
+        for i in range(0, self.rate_limit_retry):
+            # For now assuming that the method is 'GET'
+            response = request_obj.get()

```

Comment:
Assumed (for the initial code review) that the method is `GET` as there's no method in `python-http-client` such as `set_method` or `get_method` as present in `java-http-client`.

Reply:
Here are the supported HTTP verbs: https://github.com/sendgrid/python-http-client/blob/master/python_http_client/client.py#L88